tests/testthat/test-shinytest2-tm_a_regression.R

```diff
@@ -0,0 +1,135 @@
+testthat::test_that("e2e - tm_a_regerssion: data parameter and module label is passed properly", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_a_regression()
+  app$expect_no_shiny_error()
+
+  testthat::expect_equal(
+    app$get_text("#teal-main_ui-root-active_tab > li.active > a"),
+    "Regression"
+  )
+
+  encoding_dataset <- app$get_text("#teal-main_ui-root-regression .help-block")
+  testthat::expect_match(encoding_dataset, "Dataset:[\n ]*CO2", all = FALSE)
+
+  app$stop()
+})
```

Comment:
@vedhav @averissimo what do you think about moving such tests into `{teal}`? 
We never really tested encodings `.help-block`. 

I am also testing if module label is passed correctly, but I think modules' labels are tested in https://github.com/insightsengineering/teal/blob/main/tests/testthat/test-shinytest2-modules.R#L91-L118

Reply:
I think this lives well here as an e2e test. In practice, we're testing if everything underneath works, that is the coordination between `{teal.widgets}`, `{teal.transform}` and `{teal} (`standard_layout`, `data_extract_spec` and `init` respectively)

`{teal}` doesn't know about `{teal.transform}` so if we want to move this, it would have to be to `{teal.transform}`, right?

Reply:
got it. lets keep in here. thanks for your thoughts : )